src/SmartComponents/ProviderPage/providerForm.js

```diff
@@ -22,7 +22,7 @@ export const providerForm = {
 import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
 export const wizardForm = {
     initialValues: {
-        role: 'kubernetes',
+        role: 'kubernetes', // 'aws' for AWS EC2
```

Comment:
Can this be encoded into the source_type schema ?

Reply:
I used the initialValues to pass in the role for the Endpoint.

The problem is that the `initialValues` here are for the whole form. Not a single choice of provider. So in order to not have this hard-coded I need to find a different way to get it from the schema.

Until I find such way I added a case with the provider-specific values including the value for the `role`.